### ISSUE TITLE
Decreasing entropy factor in loss function

### ISSUE THREAD

**Issue by DoubleAMF** [greydanus/baby-a3c#5]
Hi @greydanus ,

I'm wondering is there a specific reason why you've added a term which decreases the entropy in the loss function. From other implementations of A3C I've seen, a factor to increase the entropy is added instead with the factor being reduced over time. My understanding is that preserving a small amount of entropy helps by encouraging exploration.

Many Thanks,
Akmal Bakar

**Comment by thibaultcalvayrac** [comment]
You are right, the idea is to increase entropy, not decrease it. But this is what the code does. 

On [line 95](https://github.com/greydanus/baby-a3c/blob/035f07ce6cc4b5401c642e3d9543b93e668579b1/baby-a3c.py#L95), you have:
```
entropy_loss = -(-logps * torch.exp(logps)).sum()  # encourage lower entropy
```
Ignore the misleading comment: by adding this term to the loss, you actually encourage **higher** entropy.

This line can be decomposed as:
```
entropy = (-logps * torch.exp(logps)).sum()  # entropy definition
entropy_loss = -entropy  # term to minimise
```
The optimisation process minimises the loss, so to maximise the entropy, you define the `entropy_loss` as the negative entropy. Minimizing `-entropy` is equivalent to maximising `entropy`, which is exactly
what you want.

**Comment by greydanus** [comment]
Fixed the misleading comment, thanks @thibaultcalvayrac
